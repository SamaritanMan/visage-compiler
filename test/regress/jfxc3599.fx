/**
 * Regression test JFXC-3599 : Compiled bind: bound block
 *
 * @test
 * @run
 */

var v = 10;
def bs = bind {
		def sq = v * v;
		def mark = 10000;
		mark + sq
};

class jfxc3599 {
  var d = "oo";
  def b = bind {
		def spl = "{d}xx{d}";
		def dd = "{spl}---{spl}";
		"{dd}===={dd}"
  }

  function doit() {
    println(b);
    d = "@@";
    println(b);
  }
}

println(bs);
v = 5;
println(bs);

jfxc3599{}.doit();
