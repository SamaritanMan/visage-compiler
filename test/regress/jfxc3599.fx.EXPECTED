10100
10025
ooxxoo---ooxxoo====ooxxoo---ooxxoo
@@xx@@---@@xx@@====@@xx@@---@@xx@@
